src/store/wallet/wallet-epics.ts

```diff
@@ -64,34 +69,23 @@ const loadTokenBalanceEpic = (action$: Observable<Action>, state$: Observable<Ro
   action$.pipe(
     ofType(loadTokensBalancesArrayActions.submit),
     toPayload(),
-    concatMap(payload =>
-      of(payload).pipe(
-        withSelectedAccount(state$),
-        withSelectedRpcUrl(state$),
-        switchMap(([[payload, selectedAccount], selectedRpcUrl]) => {
-          if (selectedAccount.publicKeyHash === payload.publicKeyHash) {
-            return forkJoin(
-              payload.slugs.map(slug =>
-                loadAssetBalance$(selectedRpcUrl, payload.publicKeyHash, slug).pipe(map(balance => ({ slug, balance })))
-              )
-            ).pipe(
-              delay(100),
-              map(data =>
-                loadTokensBalancesArrayActions.success({
-                  publicKeyHash: payload.publicKeyHash,
-                  data: data.filter((item): item is TokenBalanceResponse => isDefined(item.balance)),
-                  selectedRpcUrl
-                })
-              )
-            );
-          }
+    withSelectedAccount(state$),
+    withSelectedRpcUrl(state$),
+    switchMap(([[payload, selectedAccount], selectedRpcUrl]) => {
+      if (selectedAccount.publicKeyHash === payload.publicKeyHash) {
```

Comment:
right now we will have `allTokens.length / 3` tzkt API requests.  
take a look how `loadTokensBalancesArrayActions.submit` is called and fix that.  

Additionally, remove this condition, as right now we should not have a queue of requests and data should be loaded only for `selectedAccount.publicKeyHash`